nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Analyze/VersionFinder.cs

```diff
@@ -133,11 +159,17 @@ public static async Task<bool> DoVersionsExistAsync(
         ILogger logger,
         CancellationToken cancellationToken)
     {
-        foreach (var packageId in packageIds)
+        foreach (var id in packageIds)
         {
-            if (!await DoesVersionExistAsync(packageId, version, nugetContext, logger, cancellationToken))
+            // Split package IDs by semicolon and check each individually.
+            var individualIds = id.Contains(';') ? id.Split(';') : new[] { id };
```

Comment:
Similar to other comments, I'd prefer if `packageIds` coming in was already split.